tianshou/data/batch.py

```diff
@@ -263,18 +264,25 @@ def __init__(self,
                  **kwargs) -> None:
         if copy:
             batch_dict = deepcopy(batch_dict)
-        if _is_batch_set(batch_dict):
-            self.stack_(batch_dict)
-        elif isinstance(batch_dict, (dict, Batch)):
-            for k, v in batch_dict.items():
-                if isinstance(v, dict) or _is_batch_set(v):
-                    self.__dict__[k] = Batch(v)
-                else:
-                    if isinstance(v, list):
-                        v = np.array(v)
-                        if not issubclass(v.dtype.type, (np.bool_, np.number)):
-                            v = v.astype(np.object)
-                    self.__dict__[k] = v
+        if batch_dict is not None:
+            if isinstance(batch_dict, (dict, Batch)):
+                for k, v in batch_dict.items():
+                    if isinstance(v, (list, tuple)):
+                        v_ = np.array(v)
```

Comment:
Why not to replace line 273 by this one ? So that introducing `v_` is unnecessary

Reply:
Hi, I run a test to check the call of ``_is_batch_set``, the code is [here](https://github.com/youkaichao/tianshou/commit/650b1e790a6185e471778444fd93932927ebb5e8). The results indicate that most calls of ``_is_batch_set`` returns ``False``.

Reply:
I run the code ``test/discrete/test_pg.py`` and record the call of ``_is_batch_set`` in three places where I mark them as init1, init2, and setattr.

Originally:
init1 called 287733 times, all return False.
init2 called 405493 times, with 397151 False.

After adding the filter ``if batch_dict is not None``, the calls changed into:
init1 called 201907 times, all return False.
init2 called 405493 times, with 397151 False.

After switching init1 and init2, the calls changed into:
init2 called 414729 times, with 397151 False.

Then we realize the actually cost is the call of _is_batch_set in init2. The major case is a list array. By chaning the code like this, no ``_is_batch_set`` is called.

Reply:
Indeed, it is a bit twisted by why not !